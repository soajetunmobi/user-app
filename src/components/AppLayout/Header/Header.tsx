import React, { FunctionComponent } from 'react'
import { routes } from 'const/routes'
import { useHistory } from 'react-router-dom'
import { useUserContext } from 'providers'

export const Header: FunctionComponent = () => {
  const {
    user: { id, email }
  } = useUserContext()

  const history = useHistory()

  const isUserLoggedIn = (): boolean => {
    return id.length !== 0 && email.length !== 0
  }

  return (
    <div className='header'>
      <div className='logo'>
        <span
          className='cursor--pointer'
          data-testid='logo-link'
          onClick={() => history.push(routes.home)}
        >
          User App
        </span>
      </div>
      <div>
        {!isUserLoggedIn() && (
          <span
            className='cursor--pointer'
            data-testid='login-link'
            onClick={() => history.push(routes.login)}
          >
            Login |{' '}
          </span>
        )}
        {isUserLoggedIn() && (
          <span
            className='cursor--pointer'
            data-testid='logout-link'
            onClick={() => window.location.reload()}
          >
            Logout |{' '}
          </span>
        )}
        <span className='cursor--pointer' onClick={() => history.push(routes.profile)}>
          Profile
        </span>
      </div>
    </div>
  )
}
