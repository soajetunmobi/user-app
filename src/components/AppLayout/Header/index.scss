.header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding-left: 20px;
    padding-right: 50px;
}

.logo {
    font-size: 2rem;
    text-align: center;
    line-height: var(--logo-height);
    height: var(--logo-height);
    margin: 0;

    & > a{
        text-decoration: none;
    }
}
