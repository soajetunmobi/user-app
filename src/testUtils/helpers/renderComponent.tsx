import React, { ReactElement, ReactNode } from 'react'
import { render } from '@testing-library/react'
import { UserContext } from 'providers'

const renderComponent = (component: ReactElement, user: User): ReactNode => {
  return render(
    <UserContext.Provider value={{ user, setUser: user => user }}>{component}</UserContext.Provider>
  )
}

export default renderComponent
