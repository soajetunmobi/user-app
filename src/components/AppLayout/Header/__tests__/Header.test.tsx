import React from 'react'
import renderer from 'react-test-renderer'
import { Header } from '../Header'
import { screen } from '@testing-library/react'
import renderComponent from 'testUtils/helpers/renderComponent'
import { MOCKED_USER } from 'testUtils/mocks/user'

describe('Header', () => {
  it('renders without crashing', () => {
    const root = renderer.create(<Header />).toJSON()
    expect(root).toMatchSnapshot()
  })

  describe('Dynamic links in header section', () => {
    it('should show login link if user is not logged in', () => {
      renderComponent(<Header />, { ...MOCKED_USER, id: '', email: '' })
      expect(screen.getByTestId('login-link')).toBeInTheDocument()
    })

    it('should show logout link if user is logged in', () => {
      renderComponent(<Header />, MOCKED_USER)
      expect(screen.getByTestId('logout-link')).toBeInTheDocument()
    })
  })
})
