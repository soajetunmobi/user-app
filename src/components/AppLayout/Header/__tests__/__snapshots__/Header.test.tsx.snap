// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<div
  className="header"
>
  <div
    className="logo"
  >
    <span
      className="cursor--pointer"
      data-testid="logo-link"
      onClick={[Function]}
    >
      User App
    </span>
  </div>
  <div>
    <span
      className="cursor--pointer"
      data-testid="login-link"
      onClick={[Function]}
    >
      Login |
       
    </span>
    <span
      className="cursor--pointer"
      onClick={[Function]}
    >
      Profile
    </span>
  </div>
</div>
`;
